File: Práctica10/PrimerParcial.java
import java.util.*;
public class LineaTeleferico {
    Public String color, tramo;
    Public int nroCabinas;
    Public int nroEmpleados;
    Public  ArrayList<String[]> empleados; // Lista de [nombre, ap_paterno, ap_materno]
    Public ArrayList<Integer> edades;
    Public ArrayList<Double> sueldos;
    public LineaTeleferico(String color, String tramo, int nroCabinas) {
        this.color = color;
        this.tramo = tramo;
        this.nroCabinas = nroCabinas;
        this.nroEmpleados = 0;
        this.empleados = new ArrayList<>();
        this.edades = new ArrayList<>();
        this.sueldos = new ArrayList<>();
    }
    public LineaTeleferico() {
        this("", "", 0);
    }
    public void agregarEmpleado(String nombre, String apPaterno, String apMaterno, int edad, double sueldo) {
        empleados.add(new String[]{nombre, apPaterno, apMaterno});
        edades.add(edad);
        sueldos.add(sueldo);
        nroEmpleados++;
    }
    public void eliminarEmpleadoPorApellido(String apellido) {
        for (int i = empleados.size() - 1; i >= 0; i--) {
            String[] emp = empleados.get(i);
            if (emp[1].equals(apellido) || emp[2].equals(apellido)) {
                empleados.remove(i);
                edades.remove(i);
                sueldos.remove(i);
                nroEmpleados--;
            }
        }
    }
    public void transferirEmpleado(int indice, LineaTeleferico otro) {
        if (indice >= 0 && indice < empleados.size()) {
            String[] emp = empleados.get(indice);
            otro.agregarEmpleado(emp[0], emp[1], emp[2], edades.get(indice), sueldos.get(indice));
            empleados.remove(indice);
            edades.remove(indice);
            sueldos.remove(indice);
            nroEmpleados--;
        }
    }
    public void mostrarEmpleadoMayorEdad() {
        if (edades.isEmpty()) {
            System.out.println("No hay empleados.");
            return;
        }
        int maxEdad = Collections.max(edades);
        int idx = edades.indexOf(maxEdad);
        String[] emp = empleados.get(idx);
        System.out.println("Empleado con mayor edad: " + String.join(" ", emp) + " - Edad: " + maxEdad);
    }
    public void mostrarEmpleadoMayorSueldo() {
        if (sueldos.isEmpty()) {
            System.out.println("No hay empleados.");
            return;
        }
        double maxSueldo = Collections.max(sueldos);
        int idx = sueldos.indexOf(maxSueldo);
        String[] emp = empleados.get(idx);
        System.out.println("Empleado con mayor sueldo: " + String.join(" ", emp) + " - Sueldo: Bs " + maxSueldo);
    }
    public static void main(String[] args) {
        LineaTeleferico t1 = new LineaTeleferico("Rojo", "Estación Central - Cementerio - 16 de Julio", 20);
        LineaTeleferico t2 = new LineaTeleferico();
        t1.agregarEmpleado("Pedro", "Rojas", "Luna", 35, 2500);
        t1.agregarEmpleado("Lucy", "Sosa", "Rios", 43, 3250);
        t1.agregarEmpleado("Ana", "Perez", "Rojas", 26, 2700);
        t1.agregarEmpleado("Saul", "Arce", "Calle", 29, 2500);
        t1.eliminarEmpleadoPorApellido("Rojas");
        t1.transferirEmpleado(0, t2);
        t1.mostrarEmpleadoMayorEdad();
        t1.mostrarEmpleadoMayorSueldo();
    }
}

File: Práctica10/PrimerParcial.py
class LineaTeleferico:
    def __init__(self, color="", tramo="", nroCabinas=0):
        self.color = color
        self.tramo = tramo
        self.nroCabinas = nroCabinas
        self.nroEmpleados = 0
        self.empleados = []       # Lista de listas [[nombre, ap_paterno, ap_materno], ...]
        self.edades = []          # Lista de edades
        self.sueldos = []         # Lista de sueldos
    def agregar_empleado(self, nombre, ap_paterno, ap_materno, edad, sueldo):
        self.empleados.append([nombre, ap_paterno, ap_materno])
        self.edades.append(edad)
        self.sueldos.append(sueldo)
        self.nroEmpleados += 1
    def eliminar_empleado_por_apellido(self, apellido):
        i = 0
        while i < len(self.empleados):
            if apellido in self.empleados[i]:
                del self.empleados[i]
                del self.edades[i]
                del self.sueldos[i]
                self.nroEmpleados -= 1
            else:
                i += 1
    def transferir_empleado(self, indice, otro_objeto):
        if 0 <= indice < self.nroEmpleados:
            otro_objeto.agregar_empleado(*self.empleados[indice], self.edades[indice], self.sueldos[indice])
            del self.empleados[indice]
            del self.edades[indice]
            del self.sueldos[indice]
            self.nroEmpleados -= 1
    def mostrar_empleado_mayor_edad(self):
        if not self.edades:
            print("No hay empleados.")
            return
        idx = self.edades.index(max(self.edades))
        print("Empleado con mayor edad:", self.empleados[idx], "- Edad:", self.edades[idx])
    def mostrar_empleado_mayor_sueldo(self):
        if not self.sueldos:
            print("No hay empleados.")
            return
        idx = self.sueldos.index(max(self.sueldos))
        print("Empleado con mayor sueldo:", self.empleados[idx], "- Sueldo:", self.sueldos[idx])
# ===============================
t1 = LineaTeleferico("Rojo", "Estación Central - Cementerio - 16 de Julio", 20)
t2 = LineaTeleferico()
# Agregar empleados al primer teleférico
t1.agregar_empleado("Pedro", "Rojas", "Luna", 35, 2500)
t1.agregar_empleado("Lucy", "Sosa", "Rios", 43, 3250)
t1.agregar_empleado("Ana", "Perez", "Rojas", 26, 2700)
t1.agregar_empleado("Saul", "Arce", "Calle", 29, 2500)
t1.eliminar_empleado_por_apellido("Rojas")
t1.transferir_empleado(0, t2)
t1.mostrar_empleado_mayor_edad()
t1.mostrar_empleado_mayor_sueldo()